tests/integrations/test_urlscan.py

```diff
@@ -0,0 +1,35 @@
+import os
+
+import pytest
+from httpx import Response
+
+from tracecat.integrations.urlscan import analyze_url
+
+
+@pytest.fixture
+def urlscan_secret(create_mock_secret) -> dict[str, str | bytes]:
+    mock_secret = create_mock_secret(
+        "urlscan", {"URLSCAN_API_KEY": os.environ["URLSCAN_API_KEY"]}
+    )
+    serialized_secret = mock_secret.model_dump()
+    return serialized_secret
+
+
+def test_analyze_url_live(urlscan_secret, respx_mock):
+    test_url = "http://google.com"
+
+    # Mock secrets manager
+    route = respx_mock.get(f'{os.environ["TRACECAT__API_URL"]}/secrets/urlscan').mock(
+        return_value=Response(status_code=200, json=urlscan_secret)
```

Comment:
@daryllimyt I'm just blocked on one final issue for the tests:
```
TypeError: Object of type bytes is not JSON serializable
```

From:
```
.mock(return_value=Response(status_code=200, json=urlscan_secret))
```
Where `urlscan_secret` is a JSON-serialized `Secret`. The issue I think is `encrypted_keys` is bytes.

We don't get this error in the `GET` secrets FastAPI endpoint because FastAPI / pydantic coerces bytes into strings before passing into the Response content (I think).

Also why `.content`: https://github.com/TracecatHQ/tracecat/blob/main/tracecat/secrets/helpers.py#L18C43-L18C63
Not `.json`?

Reply:
> Why `.content` and not `.json`
Want to leverage Pydantic to perform the model construction durectly from bytes

Reply:
> We don't get this error in the GET secrets FastAPI endpoint because FastAPI / pydantic coerces bytes into strings before passing into the Response content (I think).

We don't get this in the GET secrets is because we use the `Secret.key` property to apply encryption/decryption. The definition is [here](https://github.com/TracecatHQ/tracecat/blob/4b2783028d47aca1b8ec159b1bc38bafebf19ae0/tracecat/db.py#L117C1-L130C54) and usage is [here](https://github.com/TracecatHQ/tracecat/blob/4b2783028d47aca1b8ec159b1bc38bafebf19ae0/tracecat/api/app.py#L1441)